### DISCUSSION TITLE
Questions about 2FA

### DISCUSSION THREAD

**Discussion by Molmin** [community/community#66022] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello!

I am Molmin, a GitHub user, and I am a Chinese student.

I received an email in August requesting dual factor authentication, but when I tried to bind my phone (+86), I found that I couldn't successfully send the verification code. Then, I checked the GitHub document and found that it doesn't support phone numbers in the Chinese region.

As a student, I often need to log in to GitHub in the school computer room for development and learning. Opening 2FA is not practical because I cannot carry my phone. The only solution is to use a verification code that can only be used with a watch.

I hope you can give me some help, thank you!

**Comment by jimmunene** [comment, marked as answer]
Here's a step-by-step guide on how you can set up 2FA using a hardware token like a security key:

Get a Security Key: Purchase a hardware security key (often called a U2F key or YubiKey) from a reputable vendor. These devices are small, portable, and can be attached to your keychain.

Register the Security Key:

Log in to your GitHub account.
Go to your account settings by clicking on your profile picture in the upper-right corner and selecting "Settings."
In the left sidebar, click on "Security."
Under "Two-factor authentication," click "Set up two-factor authentication."
GitHub will prompt you to choose a 2FA method. Select "Security key."
Follow the Setup Process: GitHub will guide you through the setup process. You'll be prompted to insert your security key into a USB port.

Complete Setup: After following the on-screen instructions and successfully registering your security key, you'll be all set up with 2FA.

Now, whenever you need to log in to GitHub in your school computer room, you can use your security key instead of a mobile phone. Simply plug in the key and follow the prompts to authenticate.

This method provides you with a secure and practical way to use 2FA without relying on a mobile phone. If you have any questions or encounter any issues during the setup process, don't hesitate to ask for further assistance. We're here to help you with your GitHub experience.

Best regards,
jim munene

**Reply by dmgerman** [reply]
thank you for the answer. However, when I go to Settings. Under "Security" there is only "Code security and analysis".

Under "Password and authentication" I see "Two factor authentication". The only option is "Enable two-factor authentication" which leads to https://github.com/settings/two_factor_authentication/setup/intro which does not seem to have a way to add a yubikey. It only has "Setup authentication app" and "SMS authentication".

**Comment by tuves** [comment]
Hi @Molmin ,

I've removed your private information from [this post](link to discussion) as posting identifiable information is not allowed, per our [Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md).

**Comment by Mng12345** [comment]
Why the phone number in China is not supported?? That is obviously an much easier way to make 2FA!

**Reply by Molmin** [reply]
Maybe sending messages to phone numbers in China are often fail?
